python_modules/dagster/dagster/core/definitions/sensor_definition.py

```diff
@@ -370,20 +370,20 @@ def check_valid_run_requests(self, run_requests: List[RunRequest]):
         target_names = [target.pipeline_name for target in self._targets]
 
         if run_requests and not self._targets:
-            raise DagsterInvariantViolationError(
+            raise Exception(
                 f"Error in sensor {self._name}: Sensor evaluation function returned a RunRequest "
                 "for a sensor without a specified target. Targets can be specified by providing "
                 "a job or pipeline_name."
             )
 
         for run_request in run_requests:
             if run_request.job_name is None and has_multiple_targets:
-                raise DagsterInvariantViolationError(
+                raise Exception(
                     f"Error in sensor {self._name}: Sensor returned a RunRequest that did not "
                     f"specify job_name for the requested run. Expected one of: {target_names}"
                 )
             elif run_request.job_name and run_request.job_name not in target_names:
-                raise DagsterInvariantViolationError(
+                raise Exception(
```

Comment:
why did I change this? This is a failure triggered by a problem in user code. DagsterInvariantViolationErrors do not get caught by user code error boundaries (not sure if this was 100% intentional?) so this error breaks through the user code boundary here: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/grpc/impl.py#L235-L244

causing a failure like this to be a grpc error / UserCodeUnreachableError (undesirable) instead of a ScheduleExecutionError (desirable)

Reply:
This was not intentional.   Thanks for catching